index.js

```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
-                                          {name: 'disableTestGeneration'},
-                                          {name: 'testFailingFiles'},
-                                          {name: 'testPassingFiles'},
-                                          {name: 'testGenerator', default: StyleLinter.prototype.testGenerator},
-                                          {name: 'consoleLogger', default: StyleLinter.prototype.consoleLogger},
-                                          {name: 'linterConfig', default: {}},
-                                          {name: 'log', default: true},
-                                          {name: 'console', default: console}];
+  /**
+   * Creates a new StyleLinter instance.
+   * Options
+   * - linterConfig           (StyleLint options)
+   * - onError                (Hook when error occurs)
+   * - testGenerator          (Hook for custom test generation)
+   * - disableTestGeneration  (Disable generatation tests for all files)
+   * - testFailingFiles       (Generate tests for failing files)
+   * - testPassingFiles       (Generate tests for passing files)
+   * - log                    (Disables error logging in console)
+   * - console                (Custom console)
+   * @class
+   */
+  constructor(inputNodes, options) {
+    super(inputNodes, options);
 
-/**
- * Creates a new StyleLinter instance.
- * Options
- * - linterConfig           (StyleLint options)
- * - onError                (Hook when error occurs)
- * - testGenerator          (Hook for custom test generation)
- * - disableTestGeneration  (Disable generatation tests for all files)
- * - testFailingFiles       (Generate tests for failing files)
- * - testPassingFiles       (Generate tests for passing files)
- * - log                    (Disables error logging in console)
- * - console                (Custom console)
- * @class
- */
-function StyleLinter(inputNodes, options) {
-  this.options = options || {linterConfig:{}};
-  this.inputNodesDirectory = resolveInputDirectory(inputNodes);
-  this.ignorer = buildIgnorer();
-  for(var i = 0; i < this.availableOptions.length; i++){
-    var option = this.availableOptions[i];
-    var name = option.name;
-    var defaultValue = option.default || this[name];
-    this[name] = typeof options[name] === "undefined" ?  defaultValue : options[name];
-  }
+    this.options = options || {linterConfig:{}};
+    this.inputNodesDirectory = resolveInputDirectory(inputNodes);
+    this.ignorer = buildIgnorer();
 
-  //TODO:remove this deprecation on v1 release
-  if(typeof options['disableConsoleLogging'] !== "undefined"){
-    console.warn('"disableConsoleLogging" propety has been deprecated in favour of "log"');
-    this.log = !options['disableConsoleLogging'];
-  }
+    /* Used to extract and delete options from input hash */
+    const availableOptions = [{name: 'onError'},
+                              {name: 'disableTestGeneration'},
+                              {name: 'testFailingFiles'},
```

Comment:
if we add defaults of true, to this and testPassingFiles. then we can delete https://github.com/billybonks/broccoli-stylelint/pull/41/files#diff-168726dbe96b3ce427e7fedce31bb0bcR101

Reply:
maybe, yeah. unrelated refactoring though. I'd like to land the ES6 changes first.

Reply:
👍